.github/workflows/test.yml

```diff
@@ -52,7 +52,14 @@ jobs:
       - name: Build ibc-go
         run: GOARCH=${{ matrix.go-arch }} LEDGER_ENABLED=false make build
       - name: Build e2e
-        run: (cd e2e; GOARCH=${{ matrix.go-arch }} go build ./...)
+        run: |
+          cd e2e
+          test_dirs="$(ls -A tests)"
+          for td in $test_dirs
+          do
+            GOARCH=${{ matrix.go-arch }} go test -c "./tests/${td}"
```

Comment:
`go test -c` compiles the tests without running them. The previous command was actually only building non `_test.go` code so was only catching some errors.

Reply:
It is also only possible to run `test -c` on a single directory at a time. It's not possible to run `go test -c ./...`